tests/ui/non_send_field_in_send_ty.stderr

```diff
@@ -0,0 +1,142 @@
+error: non-`Send` field found in a `Send` struct
+  --> $DIR/non_send_field_in_send_ty.rs:10:5
+   |
+LL |     data: Vec<UnsafeCell<T>>,
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: `-D clippy::non-send-field-in-send-ty` implied by `-D warnings`
+note: type `std::vec::Vec<std::cell::UnsafeCell<T>>` doesn't implement `Send` when `RingBuffer<T>` is `Send`
+  --> $DIR/non_send_field_in_send_ty.rs:15:1
+   |
+LL | unsafe impl<T> Send for RingBuffer<T> {}
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
```

Comment:
I intentionally limited the suggestion logic to direct uses of type parameters because I don't know how to detect the correct bound for type parameters inside a struct. `T: Send` is insufficient in some cases (e.g., `Arc<T>: Send` requires `T: Send + Sync`), and showing "add `T: Send` bound" would be incorrect in such cases. Is there a compiler API to detect such bound? Otherwise, I think the current implementation is the safest option.

Another option considered is to make a suggestion like `UnsafeCell<T>: Send` bound. This would be always correct, but it suggests non-idiomatic bound in many cases (`T: Send` is usually preferred), so I chose to not give any suggestions and rely on programmers in those cases.

Reply:
Good point, I forgot the fact that implementations can also have several bounds :sweat_smile:. It's unlikely that the compiler provides an API for this, as it's not something the compiler needs. It's a good decision to not include an incomplete help message. :+1:

Then I think it's sufficient to have the solution of *adding bounds* listed in the lint documentation. You can resolve this, as it's already listed as a viable solution :upside_down_face: